File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/mappers/elements/ApiElementDataMapper.java
package com.gigigo.orchextra.core.data.mappers.elements;

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;
import com.gigigo.orchextra.core.data.dto.elements.ApiElementData;
import com.gigigo.orchextra.core.data.mappers.elementcache.ApiElementCacheMapper;
import com.gigigo.orchextra.core.domain.entities.elements.ElementData;

public class ApiElementDataMapper implements
    ExternalClassToModelMapper<ApiElementData, ElementData> {

  private final ApiElementCacheMapper apiElementCacheMapper;

  public ApiElementDataMapper(ApiElementCacheMapper apiElementCacheMapper) {
    this.apiElementCacheMapper = apiElementCacheMapper;
  }

  @Override public ElementData externalClassToModel(ApiElementData data) {
    ElementData model = new ElementData();

    model.setElement(apiElementCacheMapper.externalClassToModel(data.getElement()));

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/mappers/elements/ApiElementDataMapper.kt
package com.gigigo.orchextra.core.data.mappers.elements

import android.util.Log
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementData
import com.gigigo.orchextra.core.data.mappers.toElementCache
import com.gigigo.orchextra.core.domain.entities.elements.ElementData

class ApiElementDataMapper : ExternalClassToModelMapper<ApiElementData, ElementData> {

  override fun externalClassToModel(data: ApiElementData): ElementData {
    val time = System.currentTimeMillis()

    val model = ElementData()

    model.element = data.element.toElementCache() //apiElementCacheMapper.externalClassToModel(data.element)

    val currentTime = System.currentTimeMillis() - time
    Log.v("TT - ApiElementData", ("" + currentTime/1000))

    return model
  }
}
